### PR TITLE
FIX #14867 blank screen issue

### LINKED ISSUE #14867: Order to invoice page shows blank screen
# Bug
[*Order to invoice page shows blank screen*]

## Environment
- **Version**: [*All version, Checked with 13.0.0 alpha*]
- **OS**: [*Mac OS*]
- **Web server**: [*Apache*]
- **PHP**: [*7.4*]
- **Database**: [*MySQL 5.7.26*]
- **URL(s)**: [*http://[webroot]/fourn/commande/orderstoinvoice.php?leftmenu=*]

## Expected and actual behavior
[*List of purchase orders should be shown with checkboxes on the right side of each line*]

## Steps to reproduce the behavior
[*Just need to go to this Url http://[webroot]/fourn/commande/orderstoinvoice.php*]

## [Attached files](https://help.github.com/articles/issue-attachments) (Screenshots, screencasts, dolibarr.log, debugging informations…)
[*Screenshot 2020-09-30 at 3 35 14 PM
Screenshot 2020-09-30 at 3 27 34 PM*]

### PR DESCRIPTION
**Pull request by rabib-metroworks**
Added multiple PO to one invoice creation functionality
Screenshot 2020-10-01 at 12 06 14 PM
Screenshot 2020-10-01 at 12 06 29 PM

### CODE REVIEW

File: htdocs/fourn/commande/orderstoinvoice.php [hunk 1]
```diff
@@ -300,44 +336,38 @@
...
-	print '<input type="hidden" name="originid" value="'.GETPOST('originid').'">';
-	print '<input type="hidden" name="socid" value="'.$soc->id.'">';
+	$dateinvoice = empty($conf->global->MAIN_AUTOFILL_DATE) ? - 1 : '';
 
-	print '<table class="border centpercent">';
+	print '<form name="add" action="' . $_SERVER["PHP_SELF"] . '" method="POST">';
+	print '<input type="hidden" name="token" value="' . $_SESSION['newtoken'] . '">';
+	print '<input type="hidden" name="action" value="add">';
+	print '<input type="hidden" name="socid" value="' . $soc->id . '">' . "\n";
+	print '<input name="facnumber" type="hidden" value="provisoire">';
```
Comment (eldy) [inline htdocs/fourn/commande/orderstoinvoice.php:352]: facnumber does not exists anymore in develop. It is ref

File: htdocs/fourn/commande/orderstoinvoice.php [hunk 2]
```diff
@@ -300,44 +336,38 @@
...
-	print '<input type="hidden" name="socid" value="'.$soc->id.'">'."\n";
-	print '<input name="ref" type="hidden" value="provisoire">';
-	print '<input type="hidden" name="origin" value="'.GETPOST('origin').'">';
-	print '<input type="hidden" name="originid" value="'.GETPOST('originid').'">';
-	print '<input type="hidden" name="socid" value="'.$soc->id.'">';
+	$dateinvoice = empty($conf->global->MAIN_AUTOFILL_DATE) ? - 1 : '';
 
-	print '<table class="border centpercent">';
+	print '<form name="add" action="' . $_SERVER["PHP_SELF"] . '" method="POST">';
+	print '<input type="hidden" name="token" value="' . $_SESSION['newtoken'] . '">';
```
Comment (eldy) [inline htdocs/fourn/commande/orderstoinvoice.php:349]: Must use newToken() to add the token into a form

**Comment by eldy** [comment]
The page commande/orderstoinvoice.php has been removed into develop branch because it is a duplicate of the same feature available into page commande/list.php (with the mass action, we can do the same).
So for vendors, we should follow same road and have the feature implemented into fourn/commande/list.php and align it like commande/list.php is for customers.

I did the work in develop branch. So you can test it if you want.

From list of purchase order, select orders and choose "Bill orders".

**Comment by rabib-metroworks** [comment]
Thanks I have checked the function. It is working fine. Can we have the feature so that we can create one invoice with multiple orders?

**Comment by eldy** [comment]
When you make the "Billing order", you have a form that ask you 
1 invoice per thirdparty or 1 invoice per order. If you choose, 1 invoice per thirdparty, the same invoice will have all orders of the same vendor.

**Pull request closed without merging**